io/io/inc/TThreadMergingServer.h

```diff
@@ -0,0 +1,92 @@
+#ifndef ROOT_TThreadMergingServer
+#define ROOT_TThreadMergingServer
+
+#include "dcqueue.h"
+
+#include "TBits.h"
+#include "TFile.h"
+#include "TFileCacheWrite.h"
+#include "TFileMerger.h"
+#include "TTimeStamp.h"
+
+struct ClientInfo {
+   TFile *fFile; // This object does *not* own the file,
+                 // it will be owned by the owner of the ClientInfo.
+   TString fLocalName;
+   UInt_t fContactsCount;
+   TTimeStamp fLastContact;
+   Double_t fTimeSincePrevContact;
+
+   ClientInfo() : fFile(0), fLocalName(), fContactsCount(0), fTimeSincePrevContact(0) {}
+   ClientInfo(const char *filename, UInt_t clientId);
+
+   void Set(TFile *file);
```

Comment:
I'd expect (reading from the name) that this is a regular setter. It seems, that we are doing more work in this method eg. some non-trivial file registration. Could you pick up a better name (eg. `RegisterFile`) and add doxygen style documentation?